### PR TITLE
Added request considerations to NodeFit Feature [#636 follow up]

### PR DESCRIPTION
**Pull request by ingvagabund**
Rebasing and addressing review comments in https://github.com/kubernetes-sigs/descheduler/pull/636

### CODE REVIEW

**Review by JaneLiuL — COMMENTED** [review]
/lgtm

File: pkg/descheduler/node/node.go [hunk 1]
```diff
@@ -133,39 +194,99 @@ func IsNodeUnschedulable(node *v1.Node) bool {
...
+	return podFitsOnNode, insufficientResources
 }
 
-// PodFitsCurrentNode checks if the given pod fits on the given node if the pod
-// node selector matches the node label.
-func PodFitsCurrentNode(pod *v1.Pod, node *v1.Node) bool {
-	ok, err := utils.PodMatchNodeSelector(pod, node)
-
+// NodeAvailableResources returns resources mapped to the quanitity available on the node.
+func NodeAvailableResources(nodeIndexer podutil.GetPodsAssignedToNodeFunc, node *v1.Node, resourceNames []v1.ResourceName) (map[v1.ResourceName]*resource.Quantity, error) {
```
[the remark this answers is not in the document]
Comment (damemi) [reply pkg/descheduler/node/node.go:227]: same with `nodeRemainingResources`, not sure these 2 need to be their own functions or not?

**Review by damemi — COMMENTED** [review]
thanks for picking this up @ingvagabund! Couple nits but overall I'm ready to merge this if you are

File: pkg/descheduler/node/node.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ingvagabund) [reply pkg/descheduler/node/node.go]: I inlined nodeRemainingResources into `NodeAvailableResources` and removed it. NodeAvailableResources lowercased.

File: pkg/descheduler/strategies/topologyspreadconstraint.go [hunk 2]
```diff
@@ -285,64 +288,14 @@ func balanceDomains(
...
-// validatePodFitsOnOtherNodes performs validation based on scheduling predicates for affinity and toleration.
-// It excludes the current node because, for the sake of domain balancing only, we care about if there is any other
-// place it could theoretically fit.
-// If the pod doesn't fit on its current node, that is a job for RemovePodsViolatingNodeAffinity, and irrelevant to Topology Spreading
-func validatePodFitsOnOtherNodes(pod *v1.Pod, nodeMap map[string]*v1.Node) error {
-	// if the pod has a hard nodeAffinity/nodeSelector/toleration that only matches this node,
-	// don't bother evicting it as it will just end up back on the same node
-	// however we still account for it "being evicted" so the algorithm can complete
-	// TODO(@damemi): Since we don't order pods wrt their affinities, we should refactor this to skip the current pod
-	// but still try to get the required # of movePods (instead of just chopping that value off the slice above)
```
[the remark this answers is not in the document]
Comment (ingvagabund) [reply pkg/descheduler/strategies/topologyspreadconstraint.go:297]: Done-ish

**Review by damemi — COMMENTED** [review]
/approve
thanks for picking this up @ingvagabund

**Review by JaneLiuL — COMMENTED** [review]
/lgtm

**Pull request merged**